frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -1149,153 +1531,84 @@ function ReaderFooter:addToMainMenu(menu_items)
-                        value = container_height,
-                        value_min = 7,
-                        value_max = 98,
-                        default_value = G_defaults:readSetting("DMINIBAR_CONTAINER_HEIGHT"),
-                        ok_text = _("Set height"),
-                        title_text = _("Container height"),
-                        keep_shown_on_apply = true,
-                        callback = function(spin)
-                            self.settings.container_height = spin.value
-                            self.height = Screen:scaleBySize(self.settings.container_height)
-                            self:refreshFooter(true, true)
-                            if touchmenu_instance then touchmenu_instance:updateItems() end
+                    {
+                        text_func = function()
+                            return T(_("2 decimal places (%1)"), self:progressPercentage(2))
                         end,
-                    }
-                    UIManager:show(items_font)
-                end,
-                keep_menu_open = true,
-            },
-            {
-                text_func = function()
-                    return T(_("Container bottom margin: %1"), self.settings.container_bottom_padding)
-                end,
-                callback = function(touchmenu_instance)
-                    local SpinWidget = require("ui/widget/spinwidget")
-                    local container_bottom_padding = self.settings.container_bottom_padding
-                    local items_font = SpinWidget:new{
-                        value = container_bottom_padding,
-                        value_min = 0,
-                        value_max = 49,
-                        default_value = 1,
-                        ok_text = _("Set margin"),
-                        title_text = _("Container bottom margin"),
-                        keep_shown_on_apply = true,
-                        callback = function(spin)
-                            self.settings.container_bottom_padding = spin.value
-                            self.bottom_padding = Screen:scaleBySize(self.settings.container_bottom_padding)
-                            self:refreshFooter(true, true)
-                            if touchmenu_instance then touchmenu_instance:updateItems() end
+                        checked_func = function()
+                            return self.settings.progress_pct_format == "2"
                         end,
-                    }
-                    UIManager:show(items_font)
-                end,
-                keep_menu_open = true,
+                        callback = function()
+                            self.settings.progress_pct_format = "2"
+                            self:refreshFooter(true)
+                        end,
+                    },
+                },
             },
             {
-                text = _("Maximum width of items"),
+                text = _("Maximum length for text items"),
```

Comment:
It's not a length though, it's a width.

Reply:
> No, for text one talks about length.

And this isn't about text length. It's about width.

> For example, your book title might have 10 characters in length which in turns translates to using x% of screen width.

No, that isn't how it works. 10 characters could be 20% of screen width or they could be 80% of screen width all while being 10 characters. These are two completely different things.

Reply:
I think we are both saying the same, only in different ways. I am saying, the lenght of the text won't be truncated when there is a screen width of x% dedicated to it, only trying to fit it into 4 words. Regardless, this name is making my head in.

it was reworded to: `Max pct of screen width used for text items`

Reply:
I think, for the visual effect of it in the menu, it could just be
`Item max width` or `Item max length` or `Text items max width/length`.
The fact that it's max % of screen width does not need to be expressed that up the menu, it's quite clear when under and using the spinwidgets.

(I don't think we use the "pct" abbreviation in the UI anywhere, we may use  a plain straight "%" when needed.)

Reply:
> I think, for the visual effect of it in the menu, it could just be
> `Item max width` or `Item max length` or `Text items max width/length`.

problem with those is the same as before, they are wrong.
`Item max width`: is not the width of the item, is the width of the screen so misleading name.
`Item max length`: this is similar to what I had before, according to @Frenzie it suggests we are adjusting the length of the string (which we are not).
`Text items max width/length`: this is the, I give up option.

all I have to say is, naming this one setting is just the worst. I would smite it to pieces like Zeus smote [Phaethon](https://en.wikipedia.org/wiki/Phaethon)

Reply:
> `Item max width`: is not the width of the item, is the width of the screen so misleading name.

But at this point upper in the menu, we don't really care about the precision of the wording: it's the sign on the door that under this menu, there are stuff to tweak the width of _some_ items: it's there under that we see what items (not all!) and that it's a max width, and what the value/unit/reference for % is.

Reply:
> we don't really care about the precision of the wording

I have been repeatedly told that concise and descriptive language is paramount. You can't change the rules now, as I said `item max width` is the wrong choice since we are not adjusting the width of the item, we are adjusting the percentage screen-width which is not the same thing.

anyway, I believe @Frenzie is okay with `Max pct of screen width used for text items` and so am I.

Reply:
Reiterating the first part of my quote: _But at this point upper in the menu_, we don't really care about the precision of the wording.

And reminding @Frenzie, even if he was ok, that we don't have the litteral "pct" used anywhere in our current UI, but we use `%`.

Reply:
My concern there is just about the visual aspect of that "Configure items" submenu, trying to not have this very long "Max pct..." in the middle of a bunch of small ones "Item separator", "Alignment: center", "Container height".

Reply:
> My concern there is just about the visual aspect of that "Configure items" submenu, trying to not have this very long "Max pct..." in the middle of a bunch of small ones "Item separator", "Alignment: center", "Container height".

now it is the first option in the second page.

Reply:
Which is better and makes this less noticable indeed.
But not in landscape mode :) where we may have it stand out.

Reply:
> anyway, I believe @Frenzie is okay

I believe I said something like "should be okay," which isn't exactly the same thing. (Concretely I meant I was only commenting on the string out of context.)

<img width="45%" alt="Screenshot_20240419_160215" src="https://github.com/koreader/koreader/assets/202757/d05c1f84-0e06-4e66-9b5b-90351ceff9c3">

> You can't change the rules now, as I said `item max width` is the wrong choice since we are not adjusting the width of the item, we are adjusting the percentage screen-width which is not the same thing.

I don't think that follows. The width of the item in percent is still the width.

My point about length is broadly as follows:
<img src=https://github.com/koreader/koreader/assets/202757/f25c8c3d-089c-4c0d-b2e2-798941627624 width=40%>

Note how the width differs greatly but the length is identical.

Anyway, what is the current string?

Reply:
This is the current string.
```
Maximum width of items
```

That seems better than any of the proposed changes.

Reply:
> in your example both strings have the exact same character length **and width** (10 and 1 respectively)

They do not. The width is in pixels or centimeters or percentages. It is not in whatever "the width of 1 llllllllll".

Reply:
> > in your example both strings have the exact same character length **and width** (10 and 1 respectively)
> 
> They do not. The width is in pixels or centimeters or percentages. It is not in "the width of 1 llllllllll".

if you are measuring in number of characters (as you do in your example), the length (longer side) is 10-characters and width (shorter side) is 1-character for both. If you measure in mm (I am going to make up numbers here) the length of `llllllll` might be 15mm and it's width 5mm, for `aaaaaaa` it would be 20mm length and 4mm width (`l`s are taller than `a`s). If we do it in "percentage of screen width" we might say `llllll` uses 5% of the total screen width while `aaaaaa` uses 8% of screen width. 

and that is why this `Maximum width of items` is wrong and misleading.

Reply:
in short, what the user is selecting is the percentage of screen width assigned to the item's length.

Reply:
> if you are measuring in number of characters (as you do in your example), the length (longer side) is 10-characters and width (shorter side) is 1-character for both.

I suspect that no one has ever thought the height of text or a building is the width because it's shorter.

> `Maximum text-item size`? that removes any type of confusion

On the contrary, size can inherently mean both width and height. You'd have to say `horizontal size (better known as width)`.

Reply:
I'm perfectly happy with `Maximum width of items`, FWIW.

The fact that we're technically talking about the maximum width (as in, screen space) an item can occupy is a technicality nobody really cares about for somethign user facing.

Reply:
> You'd have to say horizontal size (better known as width).

I'm sorry but that is not necessarily true. Width (in a 2-D object) is **always** the shorter side, regardless of being horizontal or vertical. Length is always the larger side.

So for two rectangles one 4x2 and another 2x16, their width is the same: 2. Their lengths are not.

Reply:
SInce this is a menu label, you can probably disambiguate to your heart's content in a `help_text` instead of obfuscating the main label ;).

Reply:
> I'm sorry but that is not necessarily true. Width (in a 2-D object) is **always** the shorter side, regardless of being horizontal or vertical. Length is always the larger side.

And yet you'd never refer to the height of a building as its width *unless you were a dumb AI*.

Reply:
> And yet you'd never refer to the height of a building as its width unless you were a dumb AI.

What are you talking about?

Reply:
![bla](https://github.com/koreader/koreader/assets/202757/e5d71eeb-c99c-4eed-b927-713a05f04c22)

Reply:
^ That is the applicable frame of reference, with a defined top and bottom.

Reply:
But i said no one talks about the width of text, is nonsensical. Nevertheless length is always the long side and width the short size.

Reply:
I think i finally understand why we keep misunderstanding one another. I am approaching this from a geometrical perspective, you are not. You are referring to this: 

![image](https://github.com/koreader/koreader/assets/97603719/1fbe8cb1-d585-4b13-854f-9bd015029cfe)

Reply:
Anybody who's ever done anything with text talks about the width of text.
https://helpx.adobe.com/indesign/using/text-frame-fitting.html

Reply:
But that's quite irrelevant. Both the length and width of the text are a consequence of the width of the item.

It's not the maximum width of "text items" but the maximum width of "items that might grow too long" (i.e. in practice only text items).

Reply:
That's for the frame not the text. Again we were talking about different things. I now understand what you mean, sorry it took us this long to be on the same page.
However i will say, little Timmy doesn't know about width and height of a container surrounding text. That's programming talk.

Reply:
> I think i finally understand why we keep misunderstanding one another.

I understand it perfectly fine, but I think it's a contrived absurdity in context. A house has a width, a height, and a depth, not a width and a length (width and height), a width and a length (height and width), a width and a length (width and depth), and a width and a length (depth and width).

Line length is measured in average characters per line. Text length is the total number of characters (or words). Of course strictly speaking you could measure line length in centimeter, but it just wouldn't make any sense because the number of characters per inch would then be variable, which is why it isn't done that way. To keep things clear you talk about width when you're talking about width and length when you're talking about length.

Reply:
> However i will say, little Timmy doesn't know about width and height of a container surrounding text. That's programming talk.

Interestingly, only the word _length_ is part of the defining vocabulary of English (i.e., among the 2000 most frequently used words). Neither _height_ nor _width_ are, but since _high_ and _wide_ are it's relatively transparent.

Nevertheless, I think either little Timmy understands the word width perfectly fine regardless of his opinions on whether it's used correctly or little Timmy doesn't understand the word length either. ;-)

Programming talk isn't really the right word for vocabulary dating back centuries, but it could be mild jargon. Still, I think for text the number of characters or words sense of length vastly exceeds the salience of the number of centimeters or percentage points it's wide (or long if you will).

Reply:
> I think it's a contrived absurdity in context. 

But we are no longer there. This could be a great idea for a comedy sketch.

> A house has a width, a height, and a depth, not a width and a length (width and height), a width and a length (height and width), a width and a length (width and depth), and a width and a length (depth and width).

We are in agreement but if you reduce the house to a box 📦 a physical one, you have height, width and length. So it depends on context. 

Back to the matter at hand,`size` is, from a user perspective a better option then `width`. Users don't think about imaginary boxes surrounding bits of text, even I didn't think of that and look where that took us. You want this software to be for all not just people who are programmers, right?

Little Timmy, i've got your back.

Reply:
> Interestingly, only the word _length_ is part of the defining vocabulary of English (i.e., among the 2000 most frequently used words).

It is actually in the top 1000 most used words in written english. The other two are top 5000. Again, i am not saying Little Timmy would not know what width is, just perhaps he would think about it in a geometric way, the short side of something. That is what he learned in school so… not the specific jargon a software developer used. 

anyway, have you ever tried to find the hypotenuse of a right triangle whilst riding dodgems? Not as easy as it sounds.

Reply:
Size is both width and height. Length and width are better. But besides being clearer and more precise, width goes together with tap zone width, screen width, page width, content width, and page slot width. Those are all visually separated blocks on a page-like object. Besides the tap zones, those are indeed imaginary.

The longer side on most of those would often but not always be the height.

> even I didn't think of that and look where that took us.

There's never been any confusion about what it meant. Nor would there be with length of course. Whether it means letters, pixels or percentages doesn't change anything about the concept of its function.

> not just people who are programmers

A programmer might say that width, height and size are all lengths. ;-) I'd prefer to call them measures.

Reply:
GH is being really bad at showing new replies for some reason.

> Again, i am not saying Little Timmy would not know what width is, just perhaps he would think about it in a geometric way, the short side of something. That is what he learned in school so… not the specific jargon a software developer used.

I would think the first and foremost association we (as in people who went to school) have with text and length is essay assignments that ask you to write a text with a minimum length of 1000 words and a maximum length of 1200 words. Not some math jargon that doesn't relate to texts.

Reply:
Note specifically that the string under discussion is "Maximum length for *text* items". "Maximum length for text" is a matter of words or characters.

"Maximum length for items" would not necessarily imply that so strongly; it would merely be out of place.

Reply:
> I would think the first and foremost association we (as in people who went to school) have with text and length is essay assignments that ask you to write a text with a minimum length of 1000 words and a maximum length of 1200 words. Not some math jargon that doesn't relate to texts.

I was referring to `max width of items`, nothing suggests that it is text related. 

And that, length as character count is exactly what I described early-on on this debate.

I am keeping `container height` and the other one, and that is 100% programmer jargon (tbf I have no idea what to name them) so meet me half way. `max size`

Reply:
Btw, I expect no animosity developped between us, no hard feelings from my side, I found this very challenging (trying to convey ideas back and forth through messages) but I am glad we finally found some common ground.

Reply:
> I am keeping `container height` and the other one

What's a container height? Is there a reason it shouldn't simply be called statusbar height? (Or possibly something a bit like `Status bar height (including padding)` or `Status bar height with padding` if that's what container means.)

Imo programmer jargon means using weird words like fifo for simple concepts like putting newly purchased food at the back. A container doesn't fit that bill provided it contains (ahem) stuff.

> Btw, I expect no animosity developped between us

Not at all, but I do suspect the sheer quantity of messages has scared off many.

Reply:
> Imo programmer jargon means using weird words like fifo for simple concepts like putting newly purchased food at the back. A container doesn't fit that bill provided it contains (ahem) stuff.

I would imagine said container is the imaginary box that contains the status bar items, so jargon 100%. I am not suggesting any Joe wouldn't know what a container is, just that in this context they might not know is some imaginary box that well. contains the items.

`Status bar height` that works for me. the other could be `status bar padding` (currently `container bottom margin`)

Reply:
If it's called margin I imagine it's margin and not padding. And the bottom part sounds like a fairly crucial detail, otherwise it also implies padding on at least the top if not the sides as well. (I.e., status bar bottom margin.)

Reply:
wait, there is a problem. It can't be `status bar height` because status bar is the `items` bit and the `progress bar` bit combined. and that height refers to only the item's imaginary box.